### DISCUSSION TITLE
Extending tuples

### DISCUSSION THREAD

**Discussion by truegoodwill** [dotnet/csharplang#7713] [Language Ideas]
It would be great to extend tuples (by combing them or adding fields). Here's an example of how the code may look. Apologies I don't know how to create a fully-fledged language spec proposal, perhaps if you see value in the idea you might like to update my proposal. 

In my use case today as I write, extending tuples is helpful in large linq operations where I "select" a tuple with lots of fields, do a where on those fields, and then do another select to continue adding fields to the tuple.

```csharp
var ab = (A: "a", B: "b"); // a, b
var extended1 = (..ab, C:"c", D:"d"); // a, b, c, d
var extended2 = (..ab, ..(C:"c", D:"d")); // a, b, c, d

var superExtended = (..ab, ..extended1); // a, b, a, b, c, d
```

**Comment by ufcpp** [comment]
https://github.com/dotnet/csharplang/discussions/424
